src/climate_data/data.py

```diff
@@ -982,6 +982,269 @@ def load_results(
         return pd.read_parquet(path)
 
 
+class FloodingData:
+    """Class for managing the flooding data used in the project."""
+
+    def __init__(
+        self,
+        root: str | Path = cdc.FLOOD_ROOT,
+        *,
+        read_only: bool = False,
+    ) -> None:
+        self._root = Path(root)
+        self._credentials_root = self._root / "credentials"
```

Comment:
This object doesn't need credentials.  The credentials are for accessing external services (e.g. the ERA5 dbs) or storage we share with partners (e.g. Azure blobs for buildings/populations)